quiche/src/lib.rs

```diff
@@ -7984,6 +7985,95 @@ impl std::fmt::Debug for Stats {
     }
 }
 
+/// QUIC Unknown Transport Parameter
+///
+/// A QUIC transport parameter that is not specifically recognized
+/// by this implementation.
+#[derive(Clone, Debug, PartialEq)]
+pub struct UnknownTransportParameter {
+    /// ID of an unknown transport parameter
+    pub id: u64,
+    /// Original data representing the value of an unknown transport parameter
+    pub value: Vec<u8>,
+}
+
+#[cfg(feature = "qlog")]
+impl From<UnknownTransportParameter>
+    for qlog::events::quic::UnknownTransportParameter
+{
+    fn from(value: UnknownTransportParameter) -> Self {
+        Self {
+            id: value.id,
+            value: value.value,
+        }
+    }
+}
+
+/// Track unknown transport parameters, up to a limit.
+#[derive(Clone, Debug, PartialEq)]
+pub struct UnknownTransportParameters {
+    // The maximum number of bytes available to store the values of unknown
+    // transport parameters.
+    /// The space remaining for storing unknown transport parameters.
+    pub capacity: usize,
+    /// The unknown transport parameters.
+    pub parameters: Vec<UnknownTransportParameter>,
+}
+
+impl<'a> Default for UnknownTransportParameters {
+    fn default() -> Self {
+        Self {
+            capacity: Self::MAX_UNKNOWN_TRANSPORT_PARAM_SIZE,
+            parameters: vec![],
+        }
+    }
+}
+
+impl UnknownTransportParameters {
+    const MAX_UNKNOWN_TRANSPORT_PARAM_SIZE: usize = 256;
```

Comment:
I'm wondering if we make this configurable, and the default be 0. That way, an app can explicitly opt-in to TP storage. One reason this might be useful is that the server has to allocate for unauthenticated connections, and even allocatiing 256 bytes is something that would be nice to avoid when we don't want to.

If you agree, there's a similar example for `enable_dgram` https://github.com/cloudflare/quiche/blob/master/quiche/src/lib.rs#L1294

Reply:
I agree and have implemented it accordingly (in the latest version of the PR).